doc/langdef.md

```diff
@@ -1203,1342 +1193,925 @@ names are like `Europe/Paris`, `CET`, or `US/Central`.
+
+**Signatures:**
+
+*   `google.protobuf.Timestamp.getMonth() -> int` (in UTC)
+*   `google.protobuf.Timestamp.getMonth(string) -> int` (with timezone)
+
+**Examples:**
+
+```
+timestamp("2023-12-25T12:00:00Z").getMonth() // 11 (December)
+```
+
+**getSeconds** \- Get the seconds from a timestamp or duration.
+
+**Signatures:**
+
+*   `google.protobuf.Timestamp.getSeconds() -> int` (in UTC)
+*   `google.protobuf.Timestamp.getSeconds(string) -> int` (with timezone)
+*   `google.protobuf.Duration.getSeconds() -> int`
+
+**Examples:**
+
+```
+timestamp("2023-12-25T12:30:30Z").getSeconds() // 30
+duration("1m30s").getSeconds() // 30
+```
+
+#### Types and Conversions
+
+**bool** `type(bool)` \- Type denotation
+
+**Signatures:**
+
+*   `bool(bool) -> bool` (identity)
+*   `bool(string) -> bool` (type conversion)
+
+**Examples:**
+
+```
+bool(true) // true
+bool("true") // true
+bool("FALSE") // false
+```
+
+**bytes** `type(bytes)` \- Type denotation
+
+**Signatures:**
+
+*   `bytes(bytes) -> bytes` (identity)
+*   `bytes(string) -> bytes` (type conversion)
+
+**Examples:**
+
+```
+bytes("hello") // b'hello'
+```
+
+**double** `type(double)` \- Type denotation
+
+**Signatures:**
+
+*   `double(double) -> double` (identity)
+*   `double(int) -> double` (type conversion)
+*   `double(uint) -> double` (type conversion)
+*   `double(string) -> double` (type conversion)
+
+**Examples:**
+
+```
+double(3.14) // 3.14
+double(10) // 10.0
+double("3.14") // 3.14 (if successful, otherwise an error)
+```
+
+**duration**
+
+**Signatures:**
+
+*   `duration(google.protobuf.Duration) -> google.protobuf.Duration`
+*   `duration(string) -> google.protobuf.Duration`
+
+**Examples:**
+
+```
+duration("1h30m") // google.protobuf.Duration representing 1 hour and 30 minutes
+```
+
+**dyn** `type(dyn)` \- Type denotation
+
+The `dyn` types does not exist at runtime, but provides a hint to the
+type-checker to disable strong type agreement checks.
+
+**Signatures:**
+
+*   `dyn(A) -> dyn` (type conversion) (where A is any type)
+
+**Examples:**
+
+```
+dyn(123) // integer 123 marked `dyn` during type-checking
+dyn("hello") // string "hello" marked `dyn` during type-checking
+```
+
+**int** `type(int)` \- Type denotation
+
+**Signatures:**
+
+*   `int(int) -> int` (identity)
+*   `int(uint) -> int` (type conversion)
+*   `int(double) -> int` (type conversion, rounds toward zero, errors if out of
+    range)
+*   `int(string) -> int` (type conversion)
+*   `int(enum E) -> int` (type conversion)
+*   `int(google.protobuf.Timestamp) -> int` (type conversion, converts to
+    seconds since Unix epoch)
+
+**Examples:**
+
+```
+int(123) // 123
+int(3.14) // 3
+int("123") // 123 (if successful, otherwise an error)
+```
+
+**list** `type(list(dyn))` \- Type denotation
+
+**map** `type(map(dyn, dyn))` \- Type denotation
+
+**null\_type** `type(null)` \- Type denotation
+
+**string** `type(string)` \- Type denotation
+
+**Signatures:**
```

Comment:
is string(bool) -> string part of the standard?

Reply:
Good question. Seems like it should be. It wasn't in the table, but I'll check the implementations.